python/triton/backends/compiler.py

```diff
@@ -8,6 +9,139 @@
 from types import ModuleType
 
 
+class AttrsDescriptor:
+    """
+    This class handles the compile-time properties for the given function
+    parameters. Different backends can add more properties to the common ones.
+    The class contains two fields:
+
+    `arg_properties`: a dictionary containing the different compile-time properties for different
+        parameters. I.e., the dictionary will look like:
+        {
+        "prop0": (0, 2, 3)
+        "prop1": (0, 4, 5)
+        }
+        Different backend might need different properties on those paraemters to enable
+        specific optimizations. The common compile time properties contained in this class
+        are :
+        - "tt.divisibility", i.e., is the given parameter divisible by 16
+        - "tt.equal_to_1", i.e., is the given parameter an integer constant 1
+
+    `arg_value`: a dictionary containing the value of the different compile-time properties, like:
+        {
+            "prop0": val0
+            "prop1": val1
+        }
+
+    """
+    __slots__ = ('arg_properties', 'property_val', '__dict__')
```

Comment:
Actually I'm thinking something like `__slots__ = ('divisibility', 'equal_to_1', 'property_values')`. This way allow we only recognize `divisibility` and `equal_to_1` by default and allow subclasses to add more there. It also allows us to use `divisibility`/`equal_to_1`/etc directly as an attribute. wdyt?

Reply:
Done. I am not sure we need `property_values` as slot, so I left it out.